src/browser/atom-application.coffee

```diff
@@ -483,25 +487,64 @@ class AtomApplication
   # Opens up a new {AtomWindow} to run specs within.
   #
   # options -
-  #   :exitWhenDone - A Boolean that, if true, will close the window upon
+  #   :headless - A Boolean that, if true, will close the window upon
   #                   completion.
   #   :resourcePath - The path to include specs from.
   #   :specPath - The directory to load specs from.
   #   :safeMode - A Boolean that, if true, won't run specs from ~/.atom/packages
   #               and ~/.atom/dev/packages, defaults to false.
-  runSpecs: ({exitWhenDone, resourcePath, specDirectory, logFile, safeMode}) ->
+  runTests: ({headless, resourcePath, executedFrom, pathsToOpen, logFile, safeMode, timeout}) ->
     if resourcePath isnt @resourcePath and not fs.existsSync(resourcePath)
       resourcePath = @resourcePath
 
+    timeout = Number.parseFloat(timeout)
+    unless Number.isNaN(timeout)
+      timeoutHandler = ->
+        console.log "The test suite has timed out because it has been running for more than #{timeout} minutes."
+        process.exit(124) # Use the same exit code as the UNIX timeout util.
+      setTimeout(timeoutHandler, timeout * 60 * 1000)
```

Comment:
What do you think about expressing timeout in seconds instead of minutes? Seems a bit more standard.